src/include/distributed/metadata_cache.h

```diff
@@ -89,6 +89,24 @@ typedef struct
 	int *arrayOfPlacementArrayLengths;
 } DistTableCacheEntry;
 
+typedef struct DistObjectCacheEntryKey
+{
+	Oid classid;
+	Oid objid;
+	int32 objsubid;
+} DistObjectCacheEntryKey;
+
+typedef struct DistObjectCacheEntry
+{
+	/* lookup key - must be first. */
+	DistObjectCacheEntryKey key;
+
+	bool isValid;
+
+	int distributionArgIndex;
+	int colocationId;
+} DistObjectCacheEntry;
```

Comment:
Why are the rest of pg_dist_object's columns not cached?

Reply:
The other 3 should only be used for upgrades